docs/PERSISTENCE.md

```diff
@@ -0,0 +1,57 @@
+
+# Log Stream
+The log stream is an interface for clients to continuously read and append data from/to multiple plogs without worrying to handle the operations of plogs. Conceptually it is a list of plogs that only last one are unsealed and can append data. It could support different types of data services such as WAL, Indexer snapshot, etc. The log stream will provide API to read/append the data, store all the metadata information such as used plog id and their offsets. In order to handle the case that a log stream crash, we need to record these metadata as well. We store the metadata of all data services in the same partition into `Metadata Manager`, which we will describe later.
+
+## Operations
+The log stream provides few operations: create, append, read, get list of plogs, update the list of plogs (removing some plogs for GC). When the log stream creates a new plog or seals a plog, it should persist the create plog id and the sealed plog id and offset via `Metadata Manager`.
+- Create: The log stream receives no inputs. It will first create one plog: the active plog. Then, it will record the active plog id and mark this log stream to be ready to handle append and read operations.
+- Append offset 
+- Append without offset: This operation takes one input: `payload`. It will return the plog id to indicate which plog stores this payload, and the current offset of this plog. The log stream will first check whether the current plog will excess the size limit if it appends the payload to the current plog. If it will not exceed the size limit, the log stream client will append the input payload to the corresponding plog and return success. Otherwise, it will first seal the old plog, record the sealed offset of the old plog, create a new plog, record the plog id of the new plog, then write the payload to the new plog. We call this process `Plog Switch`. During this process, all the upcoming write should wait until the switching process finished in order to ensure correctness.
```

Comment:
```suggestion
- Append without offset: This operation takes one input: `payload`. It will return the plog id to indicate which plog stores this payload, and the current offset of this plog. The log stream will first check whether the current plog will exceed the size limit if it appends the payload to the current plog. If it will not exceed the size limit, the log stream client will append the input payload to the corresponding plog and return success. Otherwise, it will first seal the old plog, record the sealed offset of the old plog, create a new plog, record the plog id of the new plog, then write the payload to the new plog. We call this process `Plog Switch`. During this process, all the upcoming write should wait until the switching process finished in order to ensure correctness.
```